migrate/migrate.go

```diff
@@ -112,7 +113,11 @@ func ensureTableQuery(ctx context.Context, db *sql.DB, fn func() error) error {
 	}
 	// 42P01 is undefined_table
 	// https://www.postgresql.org/docs/9.6/static/errcodes-appendix.html
-	if pErr, ok := err.(*pq.Error); !ok || pErr.Code != "42P01" {
+	if pqErr, ok := err.(*pq.Error); ok && pqErr.Code == "42P01" {
```

Comment:
Worth making the pq error code a const? Only two references at this point, but just a thought

Reply:
Hmm. Not a bad idea, maybe a package with Postgres errors that we use in-app to hold those.

`pqerr.UndefinedTable` or something like that?

We are using 2 different drivers (pgx for switchover and data generation, pq for everything else), but it seems like they both play well in keeping the code the same format/string.

I like the idea of something like `pqerr.IsCode(pqerr.CodeUndefinedTable, err)` or simply `pqerr.IsCode("42P01", err)` that could handle the two drivers transparently. There's a lot of stuff in `maperror.go` that could be more portable/cleaner that way.

---

I think that might be better as a PR on its own though -- there are quite a few places we are doing these checks, and I'd rather not introduce a new style in this particular spot without updating the rest to match.

@m17ch @arurao What do you guys think?

Reply:
We do something similar in an API i'm working on - keeps it transparent and allows us to do any massaging of the errors we may want to do.